Scenes/Screens/RevivalShoop/RevivalShoop.gd

```diff
@@ -0,0 +1,42 @@
+extends Node2D
+
+onready var game_manager = get_parent().get_node("GameManager")
+
+const ENEMIES_PER_LEVEL_PRICE = 0
+const EARN_GOLD_PRIVE = 0
+
+func _ready():
+	game_manager.stop_gameplay()
+
+func _process(delta):
+	set_enemies_per_level_button()
+
+###########
+func set_enemies_per_level_button():
+	get_node("ColorRect/Item_1/Button1").disabled = return_enemies_per_level_access();
+
+func return_enemies_per_level_access() -> bool:
+	if get_parent().find_node("LevelManager").dwarves_per_level <= 1:
+		return false #Ulepszono do maximum
+	var result : bool = get_parent().find_node("GameData").silver_moon < ENEMIES_PER_LEVEL_PRICE
+	return result;
+
+func upgrade_enemies_per_level():
+	get_parent().find_node("LevelManager").dwarves_per_level -= 1
+###########
+
+###########
+func set_earn_gold_button():
+	get_node("ColorRect/Item_2/Button2").disabled = return_earn_gold_access();
```

Comment:
Name Button2 means nothing. Could You find better name? And using function find_node may be better idea, because position of this button in scene tree may be changed in future.

Reply:
I was in a hurry ;)